### PR TITLE
HTML toString support (rc0x03) rebased and resolved

### PR DESCRIPTION
**Pull request by teleological**
Resolved conflicts in pending PR #326

### CODE REVIEW

**Comment by teleological** [comment]
It looks like the build is failing due to a memory leak under node@0.8 when validating. Not clear how that could be related to this commit.

**Comment by teleological** [comment]
Tried updating nan to v2.1.0 to see if the fix in v2.0.9 might help. It did not.

**Comment by teleological** [comment]
Am not able to repro with node v0.8.28 on darwin-x64.

**Comment by teleological** [comment]
Please do not merge while tracing is in place. Activating GC tracing and only running the validation memory usage test has caused it to pass!

**Comment by teleological** [comment]
Please do not merge. Verifying whether tests pass under v0.8 without change.

File: src/xml_node.cc [hunk 1]
```diff
@@ -136,7 +136,47 @@ NAN_METHOD(XmlNode::ToString) {
   XmlNode *node = Nan::ObjectWrap::Unwrap<XmlNode>(info.Holder());
   assert(node);
 
-  return info.GetReturnValue().Set(node->to_string());
+  int options = 0;
+
+  if (info.Length() > 0) {
+      if (info[0]->IsBoolean()) {
+          if (info[0]->ToBoolean()->BooleanValue() == true)
```
Comment (defunctzombie) [inline src/xml_node.cc]: braces required for if statements

File: src/xml_node.cc [hunk 2]
```diff
@@ -136,7 +136,47 @@ NAN_METHOD(XmlNode::ToString) {
...
+            options |= XML_SAVE_FORMAT;
+      }else if (info[0]->IsObject()) {
```
Comment (defunctzombie) [inline src/xml_node.cc]: spaces after braces plz

**Comment by teleological** [comment]
Thank you for the review. Made the indicated changes.

File: src/xml_document.cc [hunk 3]
```diff
@@ -174,7 +176,11 @@ NAN_METHOD(XmlDocument::ToString)
 
     xmlChar* buffer = NULL;
     int len = 0;
-    xmlDocDumpFormatMemoryEnc(document->xml_obj, &buffer, &len, "UTF-8", info[0]->BooleanValue() ? 1 : 0);
+    if (document->xml_obj->type == XML_HTML_DOCUMENT_NODE) {
+        htmlDocDumpMemoryFormat(document->xml_obj, &buffer, &len, info[0]->BooleanValue() ? 1 : 0);
+    }else{
```
Comment (defunctzombie) [inline src/xml_document.cc]: space around } {

**Comment by defunctzombie** [comment]
last minor change and should be good to go!

**Comment by teleological** [comment]
Fixed that one, too. Thanks again.

**Comment by defunctzombie** [comment]
@teleological could you update the wiki docs for this so users know the feature is available and I will cut a release

**Comment by teleological** [comment]
@defunctzombie Will do. Could #307/#342 also be part of that release? I could update docs for that as well.

**Pull request merged**